core/src/main/scala/zio/logging/Logging.scala

```diff
@@ -46,31 +48,31 @@ object Logging {
 
   def make[R](
     logger: LogWriter[R],
-    rootLoggerName: Option[String] = None
+    rootLoggerName: Option[String] = None,
+    initialContext: LogContext = LogContext.empty
   ): ZLayer[R, Nothing, Logging] =
-    ZLayer.fromEffect(
-      ZIO
-        .environment[R]
-        .flatMap(env =>
-          FiberRef
-            .make(LogContext.empty)
-            .tap(_.getAndUpdateSome {
-              case ctx if rootLoggerName.isDefined =>
-                ctx.annotate(LogAnnotation.Name, rootLoggerName.toList)
-            })
-            .map { ref =>
-              new Logger[String] {
-                def locally[R1, E, A](f: LogContext => LogContext)(zio: ZIO[R1, E, A]): ZIO[R1, E, A] =
-                  ref.get.flatMap(context => ref.locally(f(context))(zio))
-
-                def log(line: => String): UIO[Unit] =
-                  ref.get.flatMap(context => logger.writeLog(context, line).provide(env))
-
-                def logContext: UIO[LogContext] = ref.get
-              }
+    ZIO
+      .environment[R]
+      .flatMap(env =>
+        FiberRef
+          .make(initialContext)
+          .tap(_.getAndUpdateSome {
+            case ctx if rootLoggerName.isDefined =>
+              ctx.annotate(LogAnnotation.Name, rootLoggerName.toList)
+          })
+          .map { ref =>
+            new Logger[String] {
+              def locally[R1, E, A](f: LogContext => LogContext)(zio: ZIO[R1, E, A]): ZIO[R1, E, A] =
+                ref.get.flatMap(context => ref.locally(f(context))(zio))
+
+              def log(line: => String): UIO[Unit] =
+                ref.get.flatMap(context => logger.writeLog(context, line).provide(env))
+
+              def logContext: UIO[LogContext] = ref.get
             }
-        )
-    )
+          }
+      )
+      .toLayer
```

Comment:
just for the record why this is cannot be `.toLayer.map(logging => Has(logging.get.derive(_ => initialContext)))` ?

Reply:
That should be good too, but it's not, maybe that's something worth exploring, but given this code:
```scala
object Examples extends zio.App {
  val correlationId = LogAnnotation[String](
    name = "correlationId",
    initialValue = "undefined-correlation-id",
    combine = (_, newValue) => newValue,
    render = identity
  )

  val logFormat = "[correlation-id = %s] %s"

  val initialContext = LogContext.empty.annotate(correlationId, UUID.randomUUID().toString)
  val env            =
    Slf4jLogger
      .make(
        logFormat = { (context, message) => logFormat.format(context(correlationId), message) },
        initialContext = initialContext
      )
//      .map(logging => Has(logging.get.derive(_ => initialContext)))

  override def run(args: List[String]) =
    (for {
      fiber <- log.locally(correlationId("1234"))(ZIO.unit).fork
      _     <- log.info("info message without correlation id")
      _     <- log.info("info message without correlation id2")
      _     <- log.info("info message without correlation id3")
      _     <- log.info("info message without correlation id4")
      _     <- fiber.join
      _     <- log.locally(correlationId("1234111")) {
                 log.info("info message with correlation id") *>
                   log.throwable("this is error", new RuntimeException("error message")).fork
               }
    } yield ExitCode.success).provideLayer(env)
}
```

1. If I provide initial context to `make` that's the result (removed boilerplate):
```
[correlation-id = 0094d077-d478-4d82-a239-a965a8c1c00e] info message without correlation id
[correlation-id = 0094d077-d478-4d82-a239-a965a8c1c00e] info message without correlation 
[correlation-id = 0094d077-d478-4d82-a239-a965a8c1c00e] info message without correlation id2
[correlation-id = 0094d077-d478-4d82-a239-a965a8c1c00e] info message without correlation
[correlation-id = 0094d077-d478-4d82-a239-a965a8c1c00e] info message without correlation id3
[correlation-id = 0094d077-d478-4d82-a239-a965a8c1c00e] info message without correlation 
[correlation-id = 0094d077-d478-4d82-a239-a965a8c1c00e] info message without correlation id4
[correlation-id = 0094d077-d478-4d82-a239-a965a8c1c00e] info message without correlation
[correlation-id = 1234111] info message with correlation id
[correlation-id = 1234111] info message with correlation 
[correlation-id = 1234111] this is error
[correlation-id = 1234111] this is error
```

2. If I use `.map(logging => Has(logging.get.derive(_ => initalContext)))` instead of passing context to `.make()` I get this:
```
[correlation-id = 4bf9434f-297c-456c-9aa3-5fff7565d949] info message without correlation id
[correlation-id = 4bf9434f-297c-456c-9aa3-5fff7565d949] info message without correlation 
[correlation-id = 4bf9434f-297c-456c-9aa3-5fff7565d949] info message without correlation id2
[correlation-id = 4bf9434f-297c-456c-9aa3-5fff7565d949] info message without correlation
[correlation-id = 4bf9434f-297c-456c-9aa3-5fff7565d949] info message without correlation id3
[correlation-id = 4bf9434f-297c-456c-9aa3-5fff7565d949] info message without correlation 
[correlation-id = 4bf9434f-297c-456c-9aa3-5fff7565d949] info message without correlation id4
[correlation-id = 4bf9434f-297c-456c-9aa3-5fff7565d949] info message without correlation
[correlation-id = 4bf9434f-297c-456c-9aa3-5fff7565d949] info message with correlation id
[correlation-id = 4bf9434f-297c-456c-9aa3-5fff7565d949] info message with correlation 
[correlation-id = 4bf9434f-297c-456c-9aa3-5fff7565d949] this is error
[correlation-id = 4bf9434f-297c-456c-9aa3-5fff7565d949] this is error
```

The correlation is always overridden

Reply:
now this makes sense when you look how `derive` is implemented
```
def derive(f: LogContext => LogContext): Logger[A] =
    new Logger[A] {
      def locally[R1, E, A1](f: LogContext => LogContext)(zio: ZIO[R1, E, A1]): ZIO[R1, E, A1] =
        self.locally(f)(zio)

      def log(line: => A): UIO[Unit] = locally(f)(self.log(line)) // this will always override

      def logContext: UIO[LogContext] = self.logContext
    }
```

I will follow up with JDG if this is semantic that we want.